SWLOR.Game.Server/Feature/ShipModuleDefinition/CapitalWeaponsModuleDefinition.cs

```diff
@@ -0,0 +1,270 @@
+                .Description(description)
+                .Type(ShipModuleType.CapitalWeapons)
+                .Texture("iit_ess_079")
+                .MaxDistance(50f)
+                .ValidTargetType(ObjectType.Creature)
+                .PowerType(ShipModulePowerType.High)
+                .RequirePerk(PerkType.OffensiveModules, 1)
+                .Recast(2f)
+                .Capacitor(5)
+                .ActivatedAction((activator, activatorShipStatus, target, targetShipStatus, moduleBonus) =>
+                {
+                    var attackRoll = Random.D10(1);
+                    var attackType = "";
+                    var attackBonus = activatorShipStatus.ThermalDamage;
+                    var attackerStat = GetAbilityScore(activator, AbilityType.Perception);
+                    var attack = Stat.GetAttack(activator, AbilityType.Perception, SkillType.Piloting, attackBonus);
+                    var defenseBonus = targetShipStatus.ThermalDefense * 2;
+                    var defense = Stat.GetDefense(target, CombatDamageType.Thermal, AbilityType.Vitality, defenseBonus);
+                    if (attackRoll < 7)
+                    {
+                        attackType = "laser";
+                    }
+                    else if (attackRoll < 9)
+                    {
+                        attackType = "ion";
+                        dmg += dmg / 3;
+                        attackBonus = activatorShipStatus.EMDamage;
+                        defenseBonus = targetShipStatus.EMDefense * 2;
+                        defense = Stat.GetDefense(target, CombatDamageType.EM, AbilityType.Vitality, defenseBonus);
+                    }
+                    else
+                    {
+                        attackBonus = activatorShipStatus.ExplosiveDamage;
+                        dmg += dmg / 4;
+                        defenseBonus = targetShipStatus.EMDefense * 2;
+                        defense = Stat.GetDefense(target, CombatDamageType.Explosive, AbilityType.Vitality, defenseBonus);
+                        if (targetShipStatus.Shield <= 4)
+                        {
+                            dmg += dmg / 4;
+                        }
+                    }
+                    var defenderStat = GetAbilityScore(target, AbilityType.Vitality);
+                    var damage = Combat.CalculateDamage(
+                        attack,
+                        dmg,
+                        attackerStat,
+                        defense,
+                        defenderStat,
+                        0);
+
+                    var chanceToHit = Space.CalculateChanceToHit(activator, target);
+                    var roll = Random.D100(1);
+                    var isHit = roll <= chanceToHit;
+
+                    var sound = EffectVisualEffect(VisualEffect.Vfx_Ship_Blast);
+                    var missile = EffectVisualEffect(VisualEffect.Mirv_StarWars_Bolt2);
+
+                    if (attackType == "laser")
```

Comment:
This method is getting very long. I would recommend splitting it up a bit. Perhaps a method for laser attacks, another for ion attacks, etc. There needs to be some organization here because it's going to be hard to maintain.